core/src/main/java/com/linecorp/armeria/server/Http1RequestDecoder.java

```diff
@@ -312,4 +342,18 @@ public void userEventTriggered(ChannelHandlerContext ctx, Object evt) throws Exc
 
         ctx.fireUserEventTriggered(evt);
     }
+
+    private void maybeInitializeKeepAliveHandler(ChannelHandlerContext ctx) {
+        final KeepAliveHandler keepAliveHandler = writer.keepAliveHandler();
+        if (keepAliveHandler != null && ctx.channel().isActive() && ctx.channel().isRegistered()) {
```

Comment:
Question: do we need to check `ctx.channel().isRegistered()`?

Reply:
The original `IdleStateHandler` checks `ctx.channel().isRegistered()`. https://github.com/netty/netty/blob/81513c3728df8add3c94fd0bdaaf9ba424925b29/handler/src/main/java/io/netty/handler/timeout/IdleStateHandler.java#L241

If this method is called by `handlerAdded()`, I think the condition is meaningful.